src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -154,4 +166,104 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
             }
         }));
     }
+
+    private async Task SubscribeDashboardLogsAsync()
+    {
+        string? dashboardResourceId = null;
+
+        try
+        {
+            using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
+            await foreach (var @event in resourceNotificationService.WatchAsync(cts.Token))
+            {
+                // We want to capture the replica of the dashboard resource. This will allow us to get the logs for the dashboard.
+                // When the resource id is not the same as the resource name, it's the replica.
+                if (@event.Resource.Name == KnownResourceNames.AspireDashboard && @event.ResourceId != @event.Resource.Name)
+                {
+                    dashboardResourceId = @event.ResourceId;
+                    break;
+                }
+            }
+        }
+        catch (OperationCanceledException)
+        {
+            distributedApplicationLogger.LogError("Failed to get the dashboard logs.");
```

Comment:
Not sure if I should log this. This could happen if dcp is unhealthy.

Reply:
I think logging it is fine. If DCP is unhealthy, what other indication do we have?

Maybe just a warning though?

Reply:
Done